osmnx/plot.py

```diff
@@ -270,14 +269,11 @@ def save_and_show(fig, ax, save, show, close, filename, file_format, dpi, axis_o
         start_time = time.time()
         plt.show()
         log('Showed the plot in {:,.2f} seconds'.format(time.time()-start_time))
-    # if show=False, close the figure if close=True to prevent display
-    elif close:
-        plt.close()
```

Comment:
The problem with ```plt.close``` is that it destroys the figure object, which defeats the purpose of sharing it.

The problem with my solution right now is that the semantics of the ```show``` parameter is a bit wacky: even if it is set to ```False``` jupyter-notebook will still show the plot. I suppose that is because it notices a figure-obj. beeing returned.

Reply:
@guenteru ok thanks.  We'll need a solution that predictably closes vs shows figures. A common use case is rendering a hundreds+ plots for image recognition model training. If you're using a notebook as your console, this requires being able to silently save figures without rendering them all in the browser. We could possibly leave the existing plt.close behavior as is, only permitting shared objects when we're showing plots. That is, if someone is asking to close the figure with this parameter, we should let them. Thoughts?

Reply:
Hi! Sorry for the late reply, I completely overlooked your ping attempts. It's been a while since I looked into the problem and since I'm a quite busy these days I'm not sure how long it will take me to provide an acceptable idea (or PR). If I manage to find a few free hours I might start working on it again. I wish you all the best for 2019!